python/prophet/serialize.py

```diff
@@ -144,6 +144,23 @@ def model_from_dict(model_dict):
     """
     model = Prophet()  # We will overwrite all attributes set in init anyway
     # Simple types
+
+    for attribute in PD_DATAFRAME:
+        if model_dict[attribute] is None:
+            setattr(model, attribute, None)
+        else:
+            df = pd.read_json(StringIO(model_dict[attribute]), typ='frame', orient='table', convert_dates=['ds'])
+            if attribute == 'train_component_cols':
+                # Special handling because of named index column
+                df.columns.name = 'component'
+                df.index.name = 'col'
+            setattr(model, attribute, df)
+    
+    # backwards compat with prophet <=1.1.4
+    if 'scaling' not in model_dict:
+        model_dict['scaling'] = 'absmax'
+        model_dict['y_min'] = 0.
```

Comment:
I might be missing something, but is there a reason we need to move this logic up here? I don't think we call / want to call any of the methods setup methods (setup_dataframe, initilize_scales) during deserialization.

Could we instead:

* add a method that handles backwards compatibility for SIMPLE_ATTRIBUTES
* put your logic in lines 160-162 in that method?

There's another PR, https://github.com/facebook/prophet/pull/2477/files, that would benefit from this too.

Reply:
Yep, you're right, we don't call it so this moving around was not necessary. 

This is a lingering change, remembered I had an iteration trying to infer the scale from `df.history` if it wasn't set (using `minmax` as default). 

Added separate method for backwards compat 👍